mobile/library/java/org/chromium/net/impl/Errors.java

```diff
@@ -81,19 +83,19 @@ public static NetError mapEnvoyMobileErrorToNetError(EnvoyFinalStreamIntel final
       return NetError.ERR_INTERNET_DISCONNECTED;
     }
 
-    // Check if negotiated_protocol is quic
-    // TODO(abeyad): Assigning all errors that happen when using HTTP/3 to QUIC_PROTOCOL_ERROR
-    // can mask the real error (DNS, upstream connection, etc). Revisit the error conversion logic.
-    if (finalStreamIntel.getUpstreamProtocol() == UpstreamHttpProtocol.HTTP3) {
-      return NetError.ERR_QUIC_PROTOCOL_ERROR;
-    }
-
     // This will only map the first matched error to a NetError code.
     for (Map.Entry<Long, NetError> entry : ENVOYMOBILE_ERROR_TO_NET_ERROR.entrySet()) {
       if ((responseFlag & entry.getKey()) != 0) {
         return entry.getValue();
       }
     }
+
+    // Check if negotiated_protocol is quic
+    // TODO(abeyad): Assigning all errors that happen when using HTTP/3 to QUIC_PROTOCOL_ERROR
+    // can mask the real error (DNS, upstream connection, etc). Revisit the error conversion logic.
+    if (finalStreamIntel.getUpstreamProtocol() == UpstreamHttpProtocol.HTTP3) {
+      return NetError.ERR_QUIC_PROTOCOL_ERROR;
```

Comment:
if we do this, what are the situations when ENVOYMOBILE_ERROR_TO_NET_ERROR won't have an entry for the error, so we'll fall back to QUIC protocol error?

Reply:
Everything in https://github.com/envoyproxy/envoy/blob/afd019b08de39d4ab864b626af2c839c74bc6c48/envoy/stream_info/stream_info.h#L39 is a possible response flag value, and we only map it to a small subset: https://github.com/envoyproxy/envoy/blob/afd019b08de39d4ab864b626af2c839c74bc6c48/mobile/library/java/org/chromium/net/impl/Errors.java#L33.  So I agree, a lot can still fall under QUIC_PROTOCOL_ERROR or other.  And yeah it's good to have the response flags in the details too